zingolib/src/wallet/transaction_records_by_id.rs

```diff
@@ -376,6 +365,51 @@ impl TransactionRecordsById {
 
         self.invalidate_transactions(txids_to_remove);
     }
+
+    /// Note this method is INCORRECT in the case of a 0-value, 0-fee transaction from the
+    /// Creating Capability.  Such a transaction would violate ZIP317, but could exist in
+    /// the Zcash protocol
+    ///  TODO:   Test and handle 0-value, 0-fee transaction
+    pub(crate) fn transaction_kind(&self, query_record: &TransactionRecord) -> TransactionKind {
+        let mut sapling_spends: Vec<&SaplingNote> =
+            Vec::with_capacity(query_record.spent_sapling_nullifiers.len());
```

Comment:
Why are you doing this explicitly?  Isn't the capacity allocation algorithm already fairly efficient?